mkinitcpio

```diff
@@ -223,10 +223,11 @@ build_image() {
             sort -z |
             LANG=C bsdtar --null -cnf - -T - |
             LANG=C bsdtar --uid 0 --gid 0 --null -cf - --format=newc @- |
-            $compress "${COMPRESSION_OPTIONS[@]}" > "$out"
+            $compress "${COMPRESSION_OPTIONS[@]}" |
+            (rm -f "$out" && umask 077 && cat > "$out")
```

Comment:
This isn't explicitly a feature of your patch, but it feels like the right time to suggest that we generate the image in the temporary $_d_workdir and only rotate it into place after successful generation. A failure anywhere in this pipeline causes the user to be left without a initramfs image. Not great.

Generating this separately also means we can avoid this awkward command group as part of the pipeline.